### PR TITLE
Improve the docs of the `filter()` arguments

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- ~Tests Added~
- [x] Documentation Added
- ~Name of contributors Added to AUTHORS.rst~
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

This PR moves the documentation of the filter options/arguments to an own page of the docs rather that having them as part of the `slice()` docstring.

### CODE REVIEW

File: docs/api/filtering.rst [hunk 1]
```diff
@@ -1,12 +1,72 @@
...
+Methods for filtering and slicing an :class:`IamDataFrame`
+----------------------------------------------------------
+
+.. automethod:: pyam.IamDataFrame.filter
+   :noindex:
+
+.. automethod:: pyam.IamDataFrame.slice
+   :noindex:
+
+The **IamSlice** class
```
Comment (phackstock) [inline docs/api/filtering.rst:56]: Is that one necessary to include in the documentation for users?

File: docs/api/filtering.rst [hunk 2]
```diff
@@ -1,12 +1,72 @@
...
+----------------------
+
+This class is an auxiliary feature to streamline the implementation of the
+:meth:`IamDataFrame.filter` method.
+
+.. autoclass:: pyam.slice.IamSlice
+   :members: dimensions, time, info
+
+Filtering using a proxy :class:`IamDataFrame`
```
Comment (phackstock) [inline docs/api/filtering.rst:65]: I've been trying to imagine how that would work and I think a short example would help here.

**Review by phackstock — COMMENTED** [review]
Looks good to me in principle. I think it would be easier for the users if you had a short example of each filter. You could use the same data frame every time and filter in different ways.

**Comment by danielhuppmann** [comment]
Thanks @phackstock - agree that we should extend the docs to include examples and I'll start an issue for it to be tackled before release 3.0, but for now, my aim was to get the docs out of `core.py`.

**Review by phackstock — APPROVED** [review]
Thanks for the clarification @danielhuppmann, in this case I'd say good to merge.

File: docs/api/filtering.rst (hunk 1 above)
Comment (phackstock) [reply docs/api/filtering.rst:56]: Fair point

**Pull request merged**